packages/mermaid/src/Diagram.ts

```diff
@@ -3,26 +3,24 @@ import { log } from './logger';
 import { getDiagram, registerDiagram } from './diagram-api/diagramAPI';
 import { detectType, getDiagramLoader } from './diagram-api/detectType';
 import { extractFrontMatter } from './diagram-api/frontmatter';
-import { isDetailedError } from './utils';
-import type { DetailedError } from './utils';
+import { UnknownDiagramError } from './errors';
+import { DetailedError } from './utils';
 
 export type ParseErrorFunction = (err: string | DetailedError | unknown, hash?: any) => void;
-
 export class Diagram {
   type = 'graph';
   parser;
   renderer;
   db;
-  private detectTypeFailed = false;
-  constructor(public txt: string, parseError?: ParseErrorFunction) {
+  private detectError?: UnknownDiagramError;
+  constructor(public text: string) {
+    this.text += '\n';
```

Comment:
Why is a newline added to the diagram-code? Is it required to make the parsing to work?

Reply:
I think so, it was already there. I only moved it up.